dev-client/src/model/soilId/soilIdSlice.ts

```diff
@@ -115,6 +118,22 @@ const soilIdSlice = createSlice({
     },
   },
   extraReducers: builder => {
+    builder.addCase(pushSoilData.fulfilled, (state, action) => {
+      applySyncActionResults(
+        /*
+         * type-cast here bc the soilData field is more permissive than the results object
+         * (it allows undefined values). this is safe since we aren't reading anything from
+         * the prior data.
+         */
+        state.soilData as Record<string, SoilData>,
```

Comment:
I thought about this -- ultimately I think it's stranger that the soil ID slice supports `undefined` at all; I wouldn't want that to cascade down to future work on syncing other data types. If there's a change to be made IMO it's removing the option in the slice data. I don't think it's actually used anywhere and the pattern doesn't appear in any other stores (correct me if I'm wrong)

Reply:
i think it's there because if you type it as `Record<string, SoilData>`, and then you ask for an ID which doesn't exist (e.g. because you're on the screen of a site which got deleted in a pull), it'll be typed as `SoilData` when it's actually undefined. really i wish we could just have a proper map type where you can put the type param as `SoilData` and then the `get` method returns `SoilData | undefined`, but seeing it in this context i think i agree with you that typing it as `SoilData | undefined` everywhere probably creates more problems than it solves :weary: